apps/src/aiComponentLibrary/chatMessage/utils.ts

```diff
@@ -0,0 +1,37 @@
+import {commonI18n} from '@cdo/apps/types/locale';
+import {AiInteractionStatus as Status} from '@cdo/generated-scripts/sharedConstants';
+
+import {Role} from './types';
+
+export const getChatMessageDisplayText = (
+  status: string,
+  role: Role,
+  chatMessageText: string,
+  showProfaneUserMessage: boolean
+): string => {
+  switch (status) {
+    case Status.OK:
+    case Status.UNKNOWN:
+      return chatMessageText;
+    case Status.PROFANITY_VIOLATION:
+      if (role === Role.ASSISTANT) {
+        return commonI18n.aiChatInappropriateModelMessage();
```

Comment:
Similarly, we'll likely want to revisit these strings and role constants and decide if they should be formally shared across AI projects, or keep them project-specific.

Reply:
Noted!